models/facility.py

```diff
@@ -130,11 +140,105 @@ def remove_staff(self, old_staff):
         Return True if succesfull else False
         """
         if old_staff.id in self.person['staff']:
+            self.delete_from_db(old_staff)
             del self.person['staff'][old_staff.id]
             return True
         return False
 
+    def save_state(self, database_name="default.db", over_write=False, up=" "):
+        if type(database_name) != str:
+            raise TypeError
```

Comment:
use a more elegant way to handle errors so that the program does not terminate but gives the user a relevant message